### PR TITLE
Fixed #26029 -- Allowed configuring custom file storage backends.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-26029

### CODE REVIEW

File: docs/howto/custom-file-storage.txt [hunk 1]
```diff
@@ -116,3 +116,23 @@ free unique filename cannot be found, a :exc:`SuspiciousFileOperation
...
+:setting:`STORAGES` setting. This setting maps storage aliases, which are a way
+to refer to a specific storage throughout Django, to a dictionary of settings
+for that specific storage backend. The settings in the inner dictionaries are
+described fully in the :setting:`STORAGES` documentation.
+
+``example`` storage can be retrieved from the
+``django.core.files.storage.storages`` dictionary::
+
+    from django.core.files.storage import storages
+    example_storage = storages['example']
```
Comment (carltongibson) [inline docs/howto/custom-file-storage.txt:138]:
```suggestion
    example_storage = storages["example"]
```

?

File: docs/howto/custom-file-storage.txt [hunk 2]
```diff
@@ -116,3 +116,23 @@ free unique filename cannot be found, a :exc:`SuspiciousFileOperation
...
+
+The first step to using your custom storage with Django is to tell Django about
+the file storage backend you'll be using. This is done using the
+:setting:`STORAGES` setting. This setting maps storage aliases, which are a way
+to refer to a specific storage throughout Django, to a dictionary of settings
+for that specific storage backend. The settings in the inner dictionaries are
+described fully in the :setting:`STORAGES` documentation.
+
+``example`` storage can be retrieved from the
+``django.core.files.storage.storages`` dictionary::
```
Comment (carltongibson) [inline docs/howto/custom-file-storage.txt:135]:
```suggestion
Storages are then accessed by alias from from the
``django.core.files.storage.storages`` dictionary::
```

File: docs/releases/4.2.txt [hunk 3]
```diff
@@ -91,6 +91,12 @@ In-memory file storage
 The new ``django.core.files.storage.InMemoryStorage`` class provides a
 non-persistent storage useful for speeding up tests by avoiding disk access.
 
+Custom file storages
+--------------------
+
+The new :setting:`STORAGES` setting allows configuring custom file storage
+backends.
```
Comment (carltongibson) [inline docs/releases/4.2.txt:98]:
```suggestion
The new :setting:`STORAGES` setting allows configuring multiple custom file
storage backends.
```

File: docs/topics/files.txt [hunk 4]
```diff
@@ -239,3 +239,15 @@ For example::
 
     class MyModel(models.Model):
         my_file = models.FileField(storage=select_storage)
+
+In order to set a storage defined in the :setting:`STORAGES` setting you can
+use lambda function::
```
Comment (carltongibson) [inline docs/topics/files.txt:244]:
```suggestion
In order to set a storage defined in the :setting:`STORAGES` setting you can
use a lambda function::
```

**Review by carltongibson — APPROVED** [review]
Looks good. Just a few minor suggestions... 👍

**Comment by felixxm** [comment]
@carltongibson Thanks for checking :+1:

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**